%% BioMed_Central_Tex_Template_v1.06
%%                                      %
%  bmc_article.tex            ver: 1.06 %
%                                       %

%%IMPORTANT: do not delete the first line of this template
%%It must be present to enable the BMC Submission system to
%%recognise this template!!

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%          <8 June 2012>              %%
%%                                     %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.html and the instructions for   %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% additional documentclass options:
%  [doublespacing]
%  [linenumbers]   - put the line numbers on margins

%%% loading packages, author definitions

%\documentclass[twocolumn]{bmcart}% uncomment this for twocolumn layout and comment line below
\documentclass{bmcart}

%%% Load packages
%\usepackage{amsthm,amsmath}
%\RequirePackage{natbib}
%\RequirePackage{hyperref}
\usepackage[utf8]{inputenc} %unicode support
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %%
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %%
%%  submitted article.                         %%
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\def\includegraphic{}
\def\includegraphics{}



%%% Put your definitions there:
\startlocaldefs
\endlocaldefs


%%% Begin ...
\begin{document}

%%% Start of article front matter
\begin{frontmatter}

\begin{fmbox}
\dochead{Research}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Methods to compute the composite log-likelihood (CLL) of allelic
  frequencies for the detection of
  signatures of selection in diploid genomes}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Specify information, if available,       %%
%% in the form:                             %%
%%   <key>={<id1>,<id2>}                    %%
%%   <key>=                                 %%
%% Comment or delete the keys which are     %%
%% not used. Repeat \author command as much %%
%% as required.                             %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\author[
   addressref={aff1},                   % id's of addresses, e.g. {aff1,aff2}
   corref={aff1},                       % id of corresponding address, if any
   %noteref={n1},                        % id's of article notes, if any
   email={filippo.biscarini@tecnoparco.org}   % email address
]{\inits{FB}\fnm{Filippo} \snm{Biscarini}}
\author[
   addressref={aff1}, 
   email={john.RS.Smith@cambridge.co.uk}
]{\inits{NN}\fnm{Nelson} \snm{Nazzicari}}
\author[
   addressref={aff1}, 
   email={john.RS.Smith@cambridge.co.uk}
]{\inits{AS}\fnm{Alessandra} \snm{Stella}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%% Repeat \address commands as much as      %%
%% required.                                %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\address[id=aff1]{%                           % unique id
  \orgname{Department of Bioinformatics, PTP}, % university, etc
  \street{Via Einstein - Loc. Cascina Codazza},                     %
  %\postcode{26900}                                % post or zip code
  \city{Lodi},                              % city
  \cny{Italy}                                    % country
}
%\address[id=aff2]{%
%  \orgname{Marine Ecology Department, Institute of Marine Sciences Kiel},
%  \street{D\"{u}sternbrooker Weg 20},
%  \postcode{24105}
%  \city{Kiel},
%  \cny{Germany}
%}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter short notes here                   %%
%%                                          %%
%% Short notes will be after addresses      %%
%% on first page.                           %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{artnotes}
%\note{Sample of title note}     % note to the article
%\note[id=n1]{Equal contributor} % note, connected to author
\end{artnotes}

\end{fmbox}% comment this for two column layout

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% Please refer to the Instructions for     %%
%% authors on http://www.biomedcentral.com  %%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{abstractbox}

\begin{abstract} % abstract
\parttitle{First part title} %if any
Text for this section.

\parttitle{Second part title} %if any
Text for this section.
\end{abstract}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The keywords begin here                  %%
%%                                          %%
%% Put each keyword in separate \kwd{}.     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{keyword}
\kwd{composite log-likelihood}
\kwd{signatures of selection}
\kwd{diploid genomes}
\end{keyword}

% MSC classifications codes, if any
%\begin{keyword}[class=AMS]
%\kwd[Primary ]{}
%\kwd{}
%\kwd[; secondary ]{}
%\end{keyword}

\end{abstractbox}
%
%\end{fmbox}% uncomment this for twcolumn layout

\end{frontmatter}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% Please refer to the instructions for     %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%% start of article main body
% <put your article body there>

%%%%%%%%%%%%%%%%
%% Background %%
%%%%%%%%%%%%%%%%
\section*{Background}
Selection, both natural and artificial, is one of the major forces that
can shape the genome of living organisms and change allele frequencies.
A mutation that is beneficial for the adaptation of an organism to its
environment, or that is of agricultural or industrial interest, tends to increase in
frequency in the population, together with neighbouring genomic regions
which tend to be hitch-hiked in the process (\cite{braverman1995hitchhiking}). 
Through the last decade, the on-going genomic revolution has been making
available hundreds of thousands of genetic markers for several animal, microbial and
plant species. By looking at the allele frequency at marker loci along
the genome of populations experiencing different selective pressures, it
is possible to identify genomic regions -and ultimately genes- involved
in processes such as domestication, adaptation, evolution and artificial
selection. There are a number of methods based on allele frequencies
that have been developed to detect such signatures of selection. A
popular method is Wright's $F_{ST}$
(\cite{wright1949genetical,nei1977f}) that has been applied to
studies in humans (\cite{akey2002interrogating}), plants
(\cite{zhao2010genomic}) and animals (\cite{kijas2012genome}).
Alternatively, the likelihood of the difference between allele
frequencies in different populations can be estimated and used to detect
the presence of signatures of selection
(\cite{nielsen2005genomic,stella2010identification}).
However, there are several ways in which such likelihoods can be
computed, and these might differ in computation requirements and
statistical properties, such as the sensitivity to detect signals of
selection or the behaviour along the margins of the dimensional space.
(add examples?)
It may therefore be of interest to investigate the statistical
properties of different estimators
for the likelihood of the difference between allele frequency, and
assess how well they are capable of detected actual signals of
selection.

In this study we evaluated 6 different ways to estimate the likelihood
of the difference in allele frequency between populations. The logarithm
of the likelihoods thus calculated were then computed and combined across sliding windows
along the genome (CLL, composite log-likelihood) in order to detect
signatures of selection. The proposed CLL approaches were compared among
them and with approaches based on $F_{ST}$ and on simple squared (Euclidean?)
distances between genotypes. All methods were first analysed numerically
along the entire dimensional space (frequencies in analysed populations
ranging from 0 to 1) and then tested with real data where strong signals
of selection are known to be present.
The results of the study are hereby presented and discussed.  

\section*{Methods}

Let's assume we have a test population \emph{T} (the population on which we want to detect
signatures of selection) and a null or reference population \emph{N} (the
contrasting population(s)). On these populations $F_{ST}$, the squared
difference of the allelic frequencies, and six different likelihood
measures of allele frequency differences have been computed and
evaluated in terms of their ability to detect signatures of selection. 
First, we present numerical details of the different calculations, then
examples with real data on which the different estimators were tested.


\subsection*{Numerical illustration}
\subsubsection*{$F_{ST}$}
The $F_{ST}$ (Wright's \emph{F} statistics for \emph{Subpopulation} vs
\emph{Total}), a.k.a. \emph{fixation index}, is a measure of the genetic
substructure in a joint population (or of the genetic
differentiation between (sub)populations). $F_{ST}$ was calculated as
described by Nei (\cite{nei1977f}); partitioning the total
expected heterozygosity ($H_T$, or gene diversity) into the
\emph{interpopulation} diversity ($D_{ST}$) and the
\emph{intrapopulation} diversity ($\overline{H}_S$, average of the
expected heterozygosity -$H_S$- in the subpopulations weighted by their size),
the $F_{ST}$ is defined as the proportion of gene diversity due to
differentiation among subpopulations:

\begin{equation} \label{eq:fstnei}
F_{st}=\frac{D_{st}}{H_T}=\frac{H_T-\overline{H}_S}{H_T}=1-\frac{\overline{H}_S}{H_T}  
\end{equation}


\begin{equation}
H_T=D_{st}+\overline{H}_S
\end{equation} 

$\overline{H}_S$ is the average expected heterozygosity within
subpopulations. Let $H_S$ be the expected heterozygosity within a single
subpopulation (with $L$ number of loci and $m$ number of alleles within locus): 

\begin{equation}
H_S=1-\frac{1}{L}\sum_{l=1}^L\sum_{i=1}^{m} p_i^2
\end{equation}

The average expected heterozygosity within the \emph{k} subpopulations is then:

\begin{equation}
\overline{H}_S=\frac{1}{k}\sum_{s=1}^k H_S
\end{equation} 

We shall now consider the expected heterozygosity in the overall
population; let the average allele frequency at locus $i$ across $k$
subpopulations be the
weighted average of individual allele frequencies:
$\overline{p}_i=\frac{1}{\sum_{s=1}^kn_s}\sum_{s=1}^k n_s p_{is}$, with
$n_s$ the subpopulation size, or, using proportions,
$\overline{p_i}=\sum_{i=1}^k w_k p_k$, with $w_k$ the proportion of
subpopulation \emph{k} on the total. The
expected heterozygosity in the total population is then:

\begin{equation}
H_T=1-\frac{1}{L}\sum_{l=1}^L \sum_{i=1}^m \overline{p}_{il}^2  
\end{equation}


$F_{st}$ measures the deviation of heterozygosity from Hardy-Weinberg
equilibrium in the total population which is due to the differentiation
among subpopulations.
The comparison between the heterozygosity expected under Hardy-Weinberg
equilibrium and the heterozygosity observed in the population gives indications with respect
to phenomena such as inbreeding, itnrogression, and admixture.

\subsubsection*{Squared difference ($D^2$)}

\subsubsection*{Binomial log-likelihood (BLL)}
Three cases: A vs B, B vs A and A vs A+B

\subsubsection*{Multiplicative log-likelihood (MLL)}
\subsubsection*{Hypergeometric log-likelihood (HGLL)}
\subsubsection*{Distribution of the differences (DIFF)}
Two cases: parametrical test, resampling method

\subsection*{``Ballons d'essai''}

Data from a population of xxx Holstein-Frisian (zzz males and xxx
females) and yyy Piedmontese (males and females?)
cattle were available. The first breed has been long selected for dairy
production, the latter for meat production. All animals were genotyped
with the Bovine 54k SNP-chip. Genotypes were edited for individual and
marker call-rate ($>95\%$ ?) and for MAF ($>0.05$ ?). Remaining missing
genotypes were imputed using (check this!). Xxx SNPs were eventually
available for analysis. We selected the zzz SNPs on chromosome 3 (BTA-3) as
working example to test the different estimators of the CLL of the
allele frequency difference between populations and the reference
methods ($F_{ST}$ and Euclidean distances) for the detection of
signatures of selection.
BTA-3 is known to host, within the gene \emph{SLC35A3} (position:
43400328-43445390 bps, approximately halfay along the chromosome), the point mutation responsible for CVM (Complex
Vertebral Malformation) in Holstein-Frisian cattle (\cite{thomsen2006missense}).  
CVM is a recessive inherited disorder that frequently causes abortion or perinatal
death in Holstein-Frisian calves. Other cattle breeds -including Piedmontese- are not affected by
the condition. A strong signal is therefore expected to be found at this
site, making this an ideally suited comparison to test different methods
for the detection of signatures of selection.

?` Also DGAT1, Myostatin, Caseins ?

\subsubsection*{Working example}
Text for this sub-sub-heading \ldots
\paragraph*{Sub-sub-sub heading for section}
sasad

\subsection*{Testing significance}

When analysing tens or hundreds of thousands of markers the problem of
multiple comparisons is incurred. This is a known statistical problem in general
(\cite{berry2007difficult}), and specifically in the field of genetic
studies (\cite{lander1994genetic,risch1996future}). As the number of
independent tests increases, the probability of obtaining at least one
false positive, under $H_0$, approaches one. This probability is a
function of the number of tests \emph{n} and of the chosen significance
threshold \emph{\(\alpha\)}: \(P(false\_positive)=1-(1-\alpha)^n\). When, for instance,
50000 SNP markers are tested along the genome in search of signatures of
selection, for $\alpha=0.01$, 500 false positive associations are expected on averege, just
by mere chance. 



\begin{itemize}
\item Bonferroni correction: very strict; implicit prior assumption that the
probability that \emph{all} tests are null ($H_0$ is true) is not small. If we believe
that all tests could be null, then aiming to make the number of false
positives zero is justifiable (\cite{wakefield2008reporting})
\item FDR
\item Permutation test
\end{itemize}

\section*{Results and discussion}
Text \ldots

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Backmatter begins here                   %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{backmatter}

\section*{Competing interests}
  The authors declare that they have no competing interests.

\section*{Author's contributions}
    Text for this section \ldots

\section*{Acknowledgements}
  Data (Marras ...)? \ldots
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%
%%  Bmc_mathpys.bst  will be used to                       %%
%%  create a .BBL file for submission.                     %%
%%  After submission of the .TEX file,                     %%
%%  you will be prompted to submit your .BBL file.         %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %%
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %%
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% if your bibliography is in bibtex format, use those commands:
\bibliographystyle{bmc-mathphys} % Style BST file
\bibliography{cll-biblio}      % Bibliography file (usually '*.bib' )

% or include bibliography directly:
% \begin{thebibliography}
% \bibitem{b1}
% \end{thebibliography}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}
  \begin{figure}[h!]
  \caption{\csentence{Sample figure title.}
      A short description of the figure content
      should go here.}
      \end{figure}

\begin{figure}[h!]
  \caption{\csentence{Sample figure title.}
      Figure legend text.}
      \end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%
\section*{Tables}
\begin{table}[h!]
\caption{Sample table title. This is where the description of the table should go.}
      \begin{tabular}{cccc}
        \hline
           & B1  &B2   & B3\\ \hline
        A1 & 0.1 & 0.2 & 0.3\\
        A2 & ... & ..  & .\\
        A3 & ..  & .   & .\\ \hline
      \end{tabular}
\end{table}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Additional Files}
  \subsection*{Additional file 1 --- Sample additional file title}
    Additional file descriptions text (including details of how to
    view the file, if it is in a non-standard format or the file extension).  This might
    refer to a multi-page table or a figure.

  \subsection*{Additional file 2 --- Sample additional file title}
    Additional file descriptions text.


\end{backmatter}
\end{document}
